src/web/snippet/acme.html

```diff
@@ -332,22 +489,448 @@ <h3>Generate New Certificate</h3>
+          $("#dnsCredentials").val("VARIOMEDIA_API_TOKEN=");
+          break;
+        case "vegadns":
+          $("#dnsCredentials").val("SECRET_VEGADNS_KEY=\nSECRET_VEGADNS_SECRET=\nVEGADNS_URL=");
+          break;
+        case "vercel":
+          $("#dnsCredentials").val("VERCEL_API_TOKEN=");
+          break;
+        case "versio":
+          $("#dnsCredentials").val("VERSIO_PASSWORD=\nVERSIO_USERNAME=");
+          break;
+        case "vinyldns":
+          $("#dnsCredentials").val("VINYLDNS_ACCESS_KEY=\nVINYLDNS_HOST=\nVINYLDNS_SECRET_KEY=");
+          break;
+        case "vkcloud":
+          $("#dnsCredentials").val("VK_CLOUD_PASSWORD=\nVK_CLOUD_PROJECT_ID=\nVK_CLOUD_USERNAME=");
+          break;
+        case "vscale":
+          $("#dnsCredentials").val("VSCALE_API_TOKEN=");
+          break;
+        case "vultr":
+          $("#dnsCredentials").val("VULTR_API_KEY=");
+          break;
+        case "webnames":
+          $("#dnsCredentials").val("WEBNAMES_API_KEY=");
+          break;
+        case "httpreq":
+          $("#dnsCredentials").val("HTTPREQ_ENDPOINT=\nHTTPREQ_MODE=\nRAW=");
+          break;
+        case "websupport":
+          $("#dnsCredentials").val("WEBSUPPORT_API_KEY=\nWEBSUPPORT_SECRET=");
+          break;
+        case "wedos":
+          $("#dnsCredentials").val("WEDOS_USERNAME=\nWEDOS_WAPI_PASSWORD=");
+          break;
+        case "yandex360":
+          $("#dnsCredentials").val("YANDEX360_OAUTH_TOKEN=\nYANDEX360_ORG_ID=");
+          break;
+        case "yandexcloud":
+          $("#dnsCredentials").val("YANDEX_CLOUD_FOLDER_ID=\nYANDEX_CLOUD_IAM_TOKEN=\ndns.admin=");
+          break;
+        case "yandex":
+          $("#dnsCredentials").val("YANDEX_PDD_TOKEN=");
+          break;
+        case "zoneee":
+          $("#dnsCredentials").val("ZONEEE_API_KEY=\nZONEEE_API_USER=");
+          break;
+        case "zonomi":
+          $("#dnsCredentials").val("ZONOMI_API_KEY=");
+          break;
+              }
+    })
 
-    // Obtain certificate from API
+    // Get filename form domains and input
+    function getFilename() {
+      var domains = $("#domainsInput").val();
+      var filename = $("#filenameInput").val();
+      if (filename.trim() == "" && !domains.includes(",")){
+        //Zoraxy filename are the matching name for domains.
+        //Use the same as domains
+        filename = domains;
+      }else if (filename != "" && !domains.includes(",")){
+        //Invalid settings. Force the filename to be same as domain
+        //if there are only 1 domain
+        filename = domains;
+      }else if (filename == "" && domains.includes(",")){
+        parent.msgbox("Filename cannot be empty for certs containing multiple domains.", false, 5000);
+        $("#obtainButton").removeClass("loading").removeClass("disabled");
+        return;
+      }
+
+      //Filename cannot contain wildcards, and wildcards are possible with DNS challenges
+      filename = filename.replace("*", "_");
```

Comment:
A backend replacement logic for * to _ is also needed. Only replacing & checking in front-end is a dangerous implementation.